app/models/enterprise.rb

```diff
@@ -37,30 +37,31 @@ class Enterprise < ApplicationRecord
                                     dependent: :destroy
   has_and_belongs_to_many :groups, join_table: 'enterprise_groups_enterprises',
                                    class_name: 'EnterpriseGroup'
-  has_many :producer_properties, foreign_key: 'producer_id'
+  has_many :producer_properties, foreign_key: 'producer_id', dependent: :destroy
   has_many :properties, through: :producer_properties
   has_many :supplied_products, class_name: 'Spree::Product',
                                foreign_key: 'supplier_id',
                                dependent: :destroy
-  has_many :distributed_orders, class_name: 'Spree::Order', foreign_key: 'distributor_id'
+  has_many :distributed_orders, class_name: 'Spree::Order', foreign_key: 'distributor_id',
+                                dependent: :nullify
   belongs_to :address, class_name: 'Spree::Address'
   belongs_to :business_address, optional: true, class_name: 'Spree::Address', dependent: :destroy
-  has_many :enterprise_fees
+  has_many :enterprise_fees, dependent: :nullify
```

Comment:
Should I add a `:restrict_with_error` or `:restrict_with_exception` to this (for now)? or just revert and TODO?

Reply:
The closest to the current behaviour is `:restrict_with_exception` because foreign keys raise an exception at the moment. But since we don't quite know how to handle this case, just leave it as it was before.